spring-web/src/main/java/org/springframework/web/util/UriBuilder.java

```diff
@@ -185,6 +186,15 @@
 	 */
 	UriBuilder queryParam(String name, Object... values);
 
+	/**
+	 * Append the given query parameter and value if and only if a value is present in the Optional.
+	 * No action will be taken, and the query parameter name will not be added, if the passed Optional is empty.
+	 * @param name the query parameter name
+	 * @param optionalValue an Optional, either empty or holding the query parameter value.
+	 * @return
+	 */
```

Comment:
This can be described in terms of delegating to the existing `queryParam(String, Object...)` or `queryParam(String, Collection<?>)` methods if and only if the Optional has a value.

Note the case of the value being a Collection. That should be explicitly addressed in the Javadoc and the implementation should ensure it delegates to the correct method.

Reply:
Done.